File: Backend/FoodieApp/userRestaurantService/src/main/java/com/example/userRestaurantService/Model/Restaurant.java
package com.example.userRestaurantService.Model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Restaurant
{
    @Id
    private String restaurantId;
    private String restaurantName;
    private String restaurantLocation;
    private List<Cuisine> cuisineList;
    private byte[] picByte;
    private String fileName;
}

File: FinalCapstoneProject/Frontend/angular/src/app/model/restaurant.ts
import { Cuisine } from "./cuisine";

export class Restaurant
{
    restaurantId:string;
    restaurantName:string;
    restaurantLocation:string;
    cuisineList:Cuisine[] = [];
    picByte: any;
    

    constructor(restaurantId:any, restaurantName:any, restaurantLocation:any, cuisineList:Cuisine[])
    {
        this.restaurantId = restaurantId;
        this.restaurantName = restaurantName;
        this.restaurantLocation = restaurantLocation;
        this.cuisineList = cuisineList;
    }
}